### PR TITLE
Add M1 blog article

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: docs/blog/_posts/2020-11-09-scala3-m1.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+Once 3.0.0 release candidate is out, no new features or breaking changes will take place on 3.0.x – it will only be updated for bug fixes. However, we are going to continue the work on making the language better and to test out our research in it. These changes will end up in Scala only as of 3.1.0.
+
+For now though, our teams are focusing the efforts on getting done with the remainder of the 40-something projects that we planned for the upcoming release back in July 2020. Many of them are already completed. Those that aren't yet – get revised and re-planned. With the current global uncertainty and with the scale of the project, cannot be certain, but we believe we have a reasonable chance of releasing 3.0.0-RC1 by Christmas.
+
+Below, you can find a short summary of the changes that took place during between the 0.27.0-RC1 and 3.0.0-M1 releases.
+
+<!--more-->
+
+# Scala.js support for Scala 3
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2020-11-09-scala3-m1.md:18]: /cc @sjrd

File: docs/blog/_posts/2020-11-09-scala3-m1.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+Implemented by PRs [#9774](https://github.com/lampepfl/dotty/pull/9774) and [#9725](https://github.com/lampepfl/dotty/pull/9725), Scala 3 now supports Scala.js. You can read more about this in the [official announcement](https://www.scala-lang.org/2020/11/03/scalajs-for-scala-3.html).
```
Comment (sjrd) [inline docs/blog/_posts/2020-11-09-scala3-m1.md:19]: I would suggest the following:
```markdown
Dotty 0.2.7.0-RC1 had introduced preliminary Scala.js support with the portable subset of Scala and native JavaScript types.
Scala 3.0.0-M1 significantly expands on that support:

* support for non-native JS types ([#9774](https://github.com/lampepfl/dotty/pull/9774)), and
* better support for other JS interop features, notably their interactions with Scala 3 features such as top-level declarations and `enum`s (e.g, [#9725](https://github.com/lampepfl/dotty/pull/9725) and [#9955](https://github.com/lampepfl/dotty/pull/9955)).

The only remaining feature of Scala.js that is not supported yet is JS exports: `@JSExport` and its siblings `@JSExportAll`, `@JSExportTopLevel` and `@JSExportStatic` are all ignored by Scala 3.0.0-M1.
Support for JS exports will come in the next release.

For related reading, you can learn more on what it took to [implement Scala.js support in Scala 3 on our blog](https://www.scala-lang.org/2020/11/03/scalajs-for-scala-3.html).

**Comment by smarter** [comment]
We should mention that JDK 14 doen't work (because of a regression fixed in https://github.com/lampepfl/dotty/pull/10135) and JDK 15 doesn't work either (because of https://github.com/lampepfl/dotty/pull/10135, will be fixed in the new scala-library release)

**Comment by bishabosha** [comment]
we should probably mention the new given syntax in patterns

**Comment by bishabosha** [comment]
I think there can be a section for the final enum API

**Comment by anatoliykmetyuk** [comment]
Updated Scala.js section, mentioned JDK we don't support yet,

File: docs/blog/_posts/2020-11-09-scala3-m1.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+```scala
+x match {
+  case Foo(y @ Bar(z)) => println(y)
+}
+```
+
+You can write:
+
+```scala
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2020-11-09-scala3-m1.md:41]:
```suggestion
```

**Pull request merged**